openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -254,6 +258,20 @@ def post(self, request, usage_key_str):
         # Save after the callback so any changes made in the callback will get persisted.
         block.save()
 
+        # Raise a signal to indicate the library block was updated
+        try:
+            library_block_key = LibraryUsageLocatorV2.from_string(usage_key_str)
+            LIBRARY_BLOCK_UPDATED.send_event(
+                library_block=LibraryBlockData(
+                    library_key=library_block_key.context_key,
+                    usage_key=library_block_key,
+                )
+            )
+
+        except InvalidKeyError:
+            # TODO raise XBLOCK_UPDATED if this API is ever used for course blocks.
```

Comment:
The XBlock API isn't supposed to know anything about specific contexts, so instead of taking an "if library, do X, else if course do Y" approach, the better option is something like this:

```python
context_impl = get_learning_context_impl(usage_key)
context_impl.send_updated_event()
```

Then add `send_updated_event` to both the abstract [`class LearningContext`](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/xblock/learning_context/learning_context.py) and the [`LibraryContext`](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content_libraries/library_context.py) subclass. Since that's fairly simple, would you mind doing it now? I know we discussed that this approach would be fine but I am hoping this is a small change.

Reply:
@bradenmacdonald that's much nicer, thank you! See [cc67395](https://github.com/openedx/edx-platform/pull/35103/commits/cc6739581ee7fc54aa78c13127dad291dfaec1c1)

Reply:
Looks nice! Just one issue with a line that's not needed, then I'll approve this :)